pkg/deploy/controller/deployment/controller.go

```diff
@@ -138,18 +138,28 @@ func (c *DeploymentController) Handle(deployment *kapi.ReplicationController) er
 		// Nothing to do in this terminal state.
 		glog.V(4).Infof("Ignoring deployment %s (status %s)", deployutil.LabelForDeployment(deployment), currentStatus)
 	case deployapi.DeploymentStatusComplete:
-		// Automatically clean up successful pods.
-		// TODO: Could probably do a lookup here to skip the delete call, but it's
-		// not worth adding yet since (delete retries will only normally occur
-		// during full a re-sync).
-		podName := deployutil.DeployerPodNameFor(deployment)
-		if err := c.podClient.deletePod(deployment.Namespace, podName); err != nil {
-			if !kerrors.IsNotFound(err) {
-				return fmt.Errorf("couldn't delete completed deployer pod %s/%s for deployment %s: %v", deployment.Namespace, podName, deployutil.LabelForDeployment(deployment), err)
+		// now list any pods in the namespace that have the specified label
+		deployerPods, err := c.podClient.getDeployerPodsFor(deployment.Namespace, deployment.Name)
+		if err != nil {
+			return fmt.Errorf("couldn't fetch deployer pods for %s after successful completion: %v", deployutil.LabelForDeployment(deployment), err)
+		}
+		glog.V(4).Infof("Deleting %d deployer pods for deployment %s", len(deployerPods), deployutil.LabelForDeployment(deployment))
+		cleanedAll := true
+		for _, deployerPod := range deployerPods {
+			if err := c.podClient.deletePod(deployerPod.Namespace, deployerPod.Name); err != nil {
+				if !kerrors.IsNotFound(err) {
+					// log the error and continue
+					glog.Errorf("couldn't delete completed deployer pod %s/%s for deployment %s: %v", deployment.Namespace, deployerPod.Name, deployutil.LabelForDeployment(deployment), err)
```

Comment:
We need to log since this is not a case of resource not existing, but rather a case where the resource deletion failed.